garnet.py

```diff
@@ -134,23 +125,23 @@ def __init__(self, width, height, add_pd, interconnect_only: bool = False,
             )
 
             # top <-> global controller ports connection
-            self.wire(self.ports.clk_in, self.global_controller.ports.clk_in)
-            self.wire(self.ports.reset_in,
-                      self.global_controller.ports.reset_in)
-            self.wire(self.ports.jtag, self.global_controller.ports.jtag)
-            self.wire(self.ports.axi4_slave,
-                      self.global_controller.ports.axi4_slave)
-            self.wire(self.ports.interrupt,
-                      self.global_controller.ports.interrupt)
-            self.wire(self.ports.cgra_running_clk_out,
-                      self.global_controller.ports.clk_out)
+            magma.wire(io.clk_in, global_controller.clk_in)
```

Comment:
this is correct syntax. but we like to use a new style of wiring: instead of `m.wire(x, y)` we do `dst @= src`. the only difference is that the order matters now, but it's more readable.

also, eventually we can probably abstract these calls into a `wrap` function that looks like

```python
def wrap(outer, inner):
    for name, port in inner.interface.ports.items():
        m.wire(port, getattr(outer, name))
```

but we can do this later, let's keep it explicit for now.

Reply:
On second thought, I like it being all explicitly written out. ignore the `wrap` suggestion 😄